app/models/concerns/profile_searching.rb

```diff
@@ -148,6 +154,24 @@ def filter_by_policy(_filter, _operator, policy_or_profile_id)
         conditions: "profiles.id IN (#{any_child_profile.to_sql})"
       }
     end
+
+    def filter_by_policy_and_profile_names(_filter, operator, value)
+      ss_qb = ScopedSearch::QueryBuilder.new(scoped_search_definition, '', scoped_search_definition.profile)
+      # Passing empty block because nothing needs to happen on yield in preprocess_parameters
+      # The correct yield is invoked in to_ext_method_sql so all notification values are updated correctly
+      # See https://github.com/wvanbergen/scoped_search/blob/master/lib/scoped_search/query_builder.rb
+      values = ss_qb.preprocess_parameters(
+        scoped_search_definition.field_by_name('name'), OPERATORS[operator], value
+      ) { |x| }
```

Comment:
This could be extracted to a method, maybe into that monkeypatch (to collect bad stuff at one location) together with the `OPERATORS` like this:
```ruby
    def self.preprocess_parameters(definition, field, operator, value)
      # Long comment explaining dark magic
      qb = ScopedSearch::QueryBuilder.new(definition, '', definition.profile)
      qb.preprocess_parameters(definition.field_by_name(field), OPERATORS[operator], value) { |_| }
    end
```
And then you can call it as:
```ruby
ScopedSearch::QueryBuilder.preprocess_parameters(scoped_search_definition, 'name', operator, value)
```